src/Samples/SarifToCsv/SdkExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.CodeAnalysis.Sarif;
+using System.Collections.Generic;
+
+namespace SarifToCsv
+{
+    public static class SdkExtensions
+    {
+        public static IEnumerable<PhysicalLocation> PhysicalLocations(this Result result)
+        {
+            foreach (Location location in result.Locations)
```

Comment:
>result.Locations [](start = 42, length = 16)

this code null derefs for results with no locations. this pattern (to make basic enumeration of nested things like locations more convenient) is an interesting idea for core sdk.

Reply:
Fixed. Yeah, I think SDK users are likely to want to traverse all of various properties under the result which require intermediate foreach traversals. It'd be nice to put some helpers in to make that easier.